### ISSUE TITLE
[bug]:GPU index search slower than version 1.x

### ISSUE THREAD

**Issue by elstic** [milvus-io/knowhere#700]
### Environment
```
Machine Configuration：8c30g
knowhere branch: main
```

### Current Behavior

run :  `pytest test_ivf.py::TestIvf::test_ivf_gpu`

NQ = 10000
TOP_K = 100
nprobe : Sequential execution in this list:  [1,2,4,8,16,32,64,128,256,512] 

index:
"GPUIVFFLAT"
"GPUIVFSQ"
"GPUIVFPQ"

#### GPUIVFFLAT
param:
```
{"dim":128,"gpu_id":0,"metric_type":"L2","nlist":1024}

```

log:
```
INFO     root:test_ivf.py:301 data entities: 1000000, metric_type:L2, latency_errs:0.5
INFO     root:test_ivf.py:320 GPUIVFFLAT build time: 1.7703s @-0.2624
ERROR    root:test_ivf.py:353 nprobe1 latency degraded: expected=0.0191, actual=0.0438
INFO     root:test_ivf.py:360 nprobe=1, nq=10000, topk=100, search in 0.0438s @1.2932, recall=0.2914
ERROR    root:test_ivf.py:353 nprobe2 latency degraded: expected=0.0307, actual=0.0773
INFO     root:test_ivf.py:360 nprobe=2, nq=10000, topk=100, search in 0.0773s @1.5179, recall=0.4401
ERROR    root:test_ivf.py:353 nprobe4 latency degraded: expected=0.05, actual=0.1447
INFO     root:test_ivf.py:360 nprobe=4, nq=10000, topk=100, search in 0.1447s @1.894, recall=0.6041
ERROR    root:test_ivf.py:353 nprobe8 latency degraded: expected=0.0893, actual=0.275
INFO     root:test_ivf.py:360 nprobe=8, nq=10000, topk=100, search in 0.275s @2.0795, recall=0.7588
ERROR    root:test_ivf.py:353 nprobe16 latency degraded: expected=0.167, actual=0.5306
INFO     root:test_ivf.py:360 nprobe=16, nq=10000, topk=100, search in 0.5306s @2.1772, recall=0.8809
ERROR    root:test_ivf.py:353 nprobe32 latency degraded: expected=0.3184, actual=1.0321
INFO     root:test_ivf.py:360 nprobe=32, nq=10000, topk=100, search in 1.0321s @2.2415, recall=0.9564
ERROR    root:test_ivf.py:353 nprobe64 latency degraded: expected=0.6156, actual=2.0125
INFO     root:test_ivf.py:360 nprobe=64, nq=10000, topk=100, search in 2.0125s @2.2692, recall=0.9898
ERROR    root:test_ivf.py:353 nprobe128 latency degraded: expected=1.1958, actual=3.928
INFO     root:test_ivf.py:360 nprobe=128, nq=10000, topk=100, search in 3.928s @2.2848, recall=0.9988
ERROR    root:test_ivf.py:353 nprobe256 latency degraded: expected=2.3431, actual=7.6748
INFO     root:test_ivf.py:360 nprobe=256, nq=10000, topk=100, search in 7.6748s @2.2755, recall=0.9999
ERROR    root:test_ivf.py:353 nprobe512 latency degraded: expected=4.6045, actual=15.0453
INFO     root:test_ivf.py:360 nprobe=512, nq=10000, topk=100, search in 15.0453s @2.2675, recall=0.9999
INFO     root:test_ivf.py:364 GPUIVFFLAT search on sift recalls: [0.2914, 0.4401, 0.6041, 0.7588, 0.8809, 0.9564, 0.9898, 0.9988, 0.9999, 0.9999]
INFO     root:test_ivf.py:365 GPUIVFFLAT search on sift latency: [0.0438, 0.0773, 0.1447, 0.275, 0.5306, 1.0321, 2.0125, 3.928, 7.6748, 15.0453]
```

#### GPUIVFSQ
param:
```
 {"dim":128,"gpu_id":0,"metric_type":"L2","nlist":1024}
```
log:
```
INFO     root:test_ivf.py:301 data entities: 1000000, metric_type:L2, latency_errs:0.5
INFO     root:test_ivf.py:320 GPUIVFSQ build time: 1.583s @-0.3404
ERROR    root:test_ivf.py:353 nprobe1 latency degraded: expected=0.0123, actual=0.0296
INFO     root:test_ivf.py:360 nprobe=1, nq=10000, topk=100, search in 0.0296s @1.4065, recall=0.2914
ERROR    root:test_ivf.py:353 nprobe2 latency degraded: expected=0.0253, actual=0.0486
INFO     root:test_ivf.py:360 nprobe=2, nq=10000, topk=100, search in 0.0486s @0.9209, recall=0.44
ERROR    root:test_ivf.py:353 nprobe4 latency degraded: expected=0.0409, actual=0.0957
INFO     root:test_ivf.py:360 nprobe=4, nq=10000, topk=100, search in 0.0957s @1.3399, recall=0.6037
ERROR    root:test_ivf.py:353 nprobe8 latency degraded: expected=0.0658, actual=0.1776
INFO     root:test_ivf.py:360 nprobe=8, nq=10000, topk=100, search in 0.1776s @1.6991, recall=0.7578
ERROR    root:test_ivf.py:353 nprobe16 latency degraded: expected=0.1157, actual=0.3269
INFO     root:test_ivf.py:360 nprobe=16, nq=10000, topk=100, search in 0.3269s @1.8254, recall=0.8788
ERROR    root:test_ivf.py:353 nprobe32 latency degraded: expected=0.2175, actual=0.5954
INFO     root:test_ivf.py:360 nprobe=32, nq=10000, topk=100, search in 0.5954s @1.7375, recall=0.9519
ERROR    root:test_ivf.py:353 nprobe64 latency degraded: expected=0.3728, actual=1.131
INFO     root:test_ivf.py:360 nprobe=64, nq=10000, topk=100, search in 1.131s @2.0338, recall=0.9822
ERROR    root:test_ivf.py:353 nprobe128 latency degraded: expected=0.7265, actual=2.1943
INFO     root:test_ivf.py:360 nprobe=128, nq=10000, topk=100, search in 2.1943s @2.0204, recall=0.9892
ERROR    root:test_ivf.py:353 nprobe256 latency degraded: expected=1.4303, actual=4.2477
INFO     root:test_ivf.py:360 nprobe=256, nq=10000, topk=100, search in 4.2477s @1.9698, recall=0.9898
ERROR    root:test_ivf.py:353 nprobe512 latency degraded: expected=2.8133, actual=8.3322
INFO     root:test_ivf.py:360 nprobe=512, nq=10000, topk=100, search in 8.3322s @1.9617, recall=0.9898
INFO     root:test_ivf.py:364 GPUIVFSQ search on sift recalls: [0.2914, 0.44, 0.6037, 0.7578, 0.8788, 0.9519, 0.9822, 0.9892, 0.9898, 0.9898]
INFO     root:test_ivf.py:365 GPUIVFSQ search on sift latency: [0.0296, 0.0486, 0.0957, 0.1776, 0.3269, 0.5954, 1.131, 2.1943, 4.2477, 8.3322]
```

#### GPUIVFPQ
param:
```
{"dim":128,"gpu_id":0,"m":8,"metric_type":"L2","nbits":8,"nlist":1024}
```
log:
```
INFO     root:test_ivf.py:301 data entities: 1000000, metric_type:L2, latency_errs:0.5
INFO     root:test_ivf.py:320 GPUIVFPQ build time: 3.4683s @-0.5593
INFO     root:test_ivf.py:360 nprobe=1, nq=10000, topk=100, search in 0.0164s @0.2615, recall=0.2476
ERROR    root:test_ivf.py:353 nprobe2 latency degraded: expected=0.0139, actual=0.0214
INFO     root:test_ivf.py:360 nprobe=2, nq=10000, topk=100, search in 0.0214s @0.5396, recall=0.3342
ERROR    root:test_ivf.py:353 nprobe4 latency degraded: expected=0.0164, actual=0.0338
INFO     root:test_ivf.py:360 nprobe=4, nq=10000, topk=100, search in 0.0338s @1.061, recall=0.4019
ERROR    root:test_ivf.py:353 nprobe8 latency degraded: expected=0.0261, actual=0.0549
INFO     root:test_ivf.py:360 nprobe=8, nq=10000, topk=100, search in 0.0549s @1.1034, recall=0.4444
ERROR    root:test_ivf.py:353 nprobe16 latency degraded: expected=0.0366, actual=0.0837
INFO     root:test_ivf.py:360 nprobe=16, nq=10000, topk=100, search in 0.0837s @1.2869, recall=0.4657
ERROR    root:test_ivf.py:353 nprobe32 latency degraded: expected=0.0539, actual=0.1307
INFO     root:test_ivf.py:360 nprobe=32, nq=10000, topk=100, search in 0.1307s @1.4249, recall=0.4736
ERROR    root:test_ivf.py:353 nprobe64 latency degraded: expected=0.0773, actual=0.2146
INFO     root:test_ivf.py:360 nprobe=64, nq=10000, topk=100, search in 0.2146s @1.7762, recall=0.4754
ERROR    root:test_ivf.py:353 nprobe128 latency degraded: expected=0.1508, actual=0.3382
INFO     root:test_ivf.py:360 nprobe=128, nq=10000, topk=100, search in 0.3382s @1.2427, recall=0.4757
ERROR    root:test_ivf.py:353 nprobe256 latency degraded: expected=0.2945, actual=0.6395
INFO     root:test_ivf.py:360 nprobe=256, nq=10000, topk=100, search in 0.6395s @1.1715, recall=0.4757
INFO     root:test_ivf.py:360 nprobe=512, nq=10000, topk=100, search in 1.2229s @0.4321, recall=0.4757
INFO     root:test_ivf.py:364 GPUIVFPQ search on sift recalls: [0.2476, 0.3342, 0.4019, 0.4444, 0.4657, 0.4736, 0.4754, 0.4757, 0.4757, 0.4757]
INFO     root:test_ivf.py:365 GPUIVFPQ search on sift latency: [0.0164, 0.0214, 0.0338, 0.0549, 0.0837, 0.1307, 0.2146, 0.3382, 0.6395, 1.2229]
```

**Comment by elstic** [comment]
@cqy123456 Help assign

**Comment by cydrain** [comment]
fixed by #713
